### PR TITLE
Hungarian taxonomy translations

### PR DESCRIPTION
**Pull request by BathoryPeter**
- labels ready
- nutriments up to Polyunsaturated fat
- more synonyms in allergens
- packagings ready
- rewording states (ready)
- fixed a mixed English-Hungarian string in nutrient_levels.result.txt (I know it is a generated file, but didn't found its source)

### CODE REVIEW

**Comment by Ban3** [comment]
> fixed a mixed English-Hungarian string in nutrient_levels.result.txt (I know it is a generated file, but didn't found its source)

It's from a perl file https://github.com/openfoodfacts/openfoodfacts-server/blob/5b089ad45e12257b4292b0c08d796cd80679740b/lib/ProductOpener/Food.pm#L2386

Looks like there isn't hu translations for salt so it's defaulting to the english.

**Comment by BathoryPeter** [comment]
> It's from a perl file

Thanks, fixed. A little bit uncommon place for a translation :)

**Comment by Ban3** [comment]
`ERROR - hu:növényi-zsírok-nélkül already is a synonym of növényi-olajok-nélkül () - cannot make a synonym of növényi-olajok for en:natural-product-without-vegetable-oils`

`növényi zsírok nélkül` defined twice as a synonym in labels.txt is creating an error when building.

File: taxonomies/labels.txt [hunk 1]
```diff
@@ -746,6 +747,7 @@ en:No thickening agent, No thickeners
 de:Ohne Verdickungsmittel
 fi:sakeuttamisaineeton, ei sakeutusaineita
 fr:Sans épaississant, Sans épaississants, Sans épaissisant, Sans épaissisants
+hu:Sűrítőanyagmentes,
```
Comment (Ban3) [inline taxonomies/labels.txt]:
```suggestion
hu:Sűrítőanyagmentes
```

File: taxonomies/labels.txt [hunk 2]
```diff
@@ -901,7 +908,8 @@ de:Hoher Omega-3-Gehalt
 es:Rico en Omega 3, Alto en Omega 3
 fi:runsaasti omega-3-rasvahappoja, paljon omega-3-rasvahappoja
 fr:Riche en Oméga-3, Riche en omega 3, Riche en acide gras omega 3
-he:עשיר באומגה 3
+he:עשיר באומגה 3,
```
Comment (Ban3) [inline taxonomies/labels.txt]:
```suggestion
he:עשיר באומגה 3
```

File: lib/ProductOpener/Food.pm [hunk 3]
```diff
@@ -2261,6 +2261,7 @@ sub mmoll_to_unit {
 		fi_synonyms => ["transrasvahappo"],
 		fr => "Acides gras trans",
 		he => "שומן טראנס - שומן בלתי רווי",
+		en => "Transz-zsírsav",
```
Comment (stephanegigandet) [inline lib/ProductOpener/Food.pm]:
```suggestion
		hu => "Transz-zsírsav",
```

**Comment by BathoryPeter** [comment]
Sorry for the mistakes.
Is there a way to test taxonomies locally? I did some attempt with `node refresh_taxonomies.js` with less success, child processes returns with error code 2 and I cannot find more detailed logs. `node check_perl.js` seems good. Also tried `npm test` but I got a ton of errors.

**Comment by Ban3** [comment]
> `node refresh_taxonomies.js` 

refresh_taxonomies.js basically wraps the `scripts/build_tags_taxonomy.pl` script. You can use it directly to build a single taxonomy and get more useful error output, for example labels:

`perl -CS -Ilib scripts/build_tags_taxonomy.pl labels 1`

Run in the project root directory. You'll need to have the project dependencies installed.

File: lib/ProductOpener/Food.pm [hunk 4]
```diff
@@ -2370,6 +2371,7 @@ sub mmoll_to_unit {
 		fi => "Natrium",
 		fr => "Sodium",
 		he => "נתרן",
+		de => "Nátrium",
```
Comment (Ban3) [inline lib/ProductOpener/Food.pm]:
```suggestion
		hu => "Nátrium",
```
Still one with a wrong language code :)

Comment (BathoryPeter) [reply lib/ProductOpener/Food.pm]: Fixed in 4ca9a5e

**Comment by VaiTon** [comment]
What a mess!

**Comment by Ban3** [comment]
> What a mess!

It happens :). @BathoryPeter probably the easiest way to recover is to recreate this pull request.

**Comment by stephanegigandet** [comment]
Hi @BathoryPeter , sorry about the mess, I don't know what happened. Could you recreate the pull request with just the files that you changed? Thank you!

**Comment by BathoryPeter** [comment]
Hi all, sorry for the noise. The weird thing is if I diff this branch with the master, I see only my own changes. I will try to reset HEAD to the last good state and move my last commit on top of that and we will see how GitHub respects is.

**Comment by VaiTon** [comment]
> 
> 
> Hi all, sorry for the noise. The weird thing is if I diff this branch with the master, I see only my own changes. I will try to reset HEAD to the last good state and move my last commit on top of that and we will see how GitHub respects is.

It's correct. You just have to see the changes you made to the code. GitHub doesn't display the actual git diff in my experience. It compares your branch with the commit you branched i suppose.

**Pull request merged**